node/bindnode/repr.go

```diff
@@ -53,8 +53,8 @@ func (w *_nodeRepr) Kind() datamodel.Kind {
 		return datamodel.Kind_Map
 	case schema.UnionRepresentation_Kinded:
 		haveIdx, _ := unionMember(w.val)
-		mtyp := w.schemaType.(*schema.TypeUnion).Members()[haveIdx]
-		return mtyp.TypeKind().ActsLike()
+		mtyp := w.schemaType.(*schema.TypeUnion).Members()[haveIdx] // TODO this is fragile: panics with index-out-of-range if the user has nil'd all fields.
```

Comment:
put a `, ok` and an error for out-of-range rather than panic?

Reply:
Nontrivial.  This method can't return errors.  It's a core interface method, and a break-the-universe-badly one if we changed its return type.